data-prepper-plugins/avro-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/avro/AvroOutputCodecConfig.java

```diff
@@ -5,18 +5,79 @@
 package org.opensearch.dataprepper.plugins.codec.avro;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Size;
 
+import java.util.List;
 
+/**
+ * Configuration class for {@link AvroOutputCodec}.
+ */
 public class AvroOutputCodecConfig {
 
     @JsonProperty("schema")
-    private final String schema;
+    private String schema;
 
-    public AvroOutputCodecConfig(String schema) {
-        this.schema = schema;
+    @Valid
+    @Size(max = 0, message = "Schema from file is not supported.")
+    @JsonProperty("schema_file_location")
+    private String fileLocation;
+
+    @JsonProperty("exclude_keys")
+    private List<String> excludeKeys;
+    @Valid
+    @Size(max = 0, message = "Schema from schema registry is not supported.")
+    @JsonProperty("schema_registry_url")
+    private String schemaRegistryUrl;
+
+    @Valid
+    @Size(max = 0, message = "Schema from file is not supported.")
+    @JsonProperty("region")
+    private String region;
+
+    @Valid
+    @Size(max = 0, message = "Schema from file is not supported.")
+    @JsonProperty("bucket_name")
+    private String bucketName;
+
+    @Valid
+    @Size(max = 0, message = "Schema from file is not supported.")
```

Comment:
The error messages on these are copied from `schema_file_location`

Reply:
Merging this for now. Please fix this @umayr-codes

Reply:
These error messages were written in accordance with Ashok's suggestion!